### PR TITLE
#3596 Manage instance ids for document sub pipeline blocks

### PR DESCRIPTION
**Pull request by BALEHOK**
## What does this PR do?

- Part of #3596 
- Adds support for add and remove block instance ids on Document sub pipelines

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

**Review by BLoe — APPROVED** [review]
Just a comment on naming for the new traverse parameter

File: src/pageEditor/utils.ts [hunk 1]
```diff
@@ -138,3 +140,84 @@ export function getDocumentPipelinePaths(block: BlockConfig): string[] {
     (block.config.body ?? []) as DocumentElement[]
   );
 }
+
+type TraversePipelineArgs = {
+  blockPipeline: BlockPipeline;
+  blockPipelinePath?: string;
+  parentNodeId?: UUID | null;
+  visitBlock: BlockAction;
+  preTraverseSubPipeline?: PreTraverseSubPipeline;
```
[the remark this answers is not in the document]
Comment (BALEHOK) [reply src/pageEditor/utils.ts:149]: There's  a discussion in another PR https://github.com/pixiebrix/pixiebrix-extension/pull/3540#discussion_r886126707

[the remark this answers is not in the document]
Comment (BLoe) [reply src/pageEditor/utils.ts:149]: I just think that, if you look at the call site, it's impossible to tell what `return false` means in that function just by looking at the name of it. I don't mean to start a larger discussion about design patterns, just giving feedback that, as another developer, this is confusing to me. If you want to leave it like this, that's fine.

[the remark this answers is not in the document]
Comment (BALEHOK) [reply src/pageEditor/utils.ts]: Makes sense. I've changed the signature a little bit.

**Pull request merged**